src/Wpf/Prism.Wpf/Extensions/DependencyObjectExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+#if HAS_WINUI
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Data;
+#else
+using System.Windows;
+using System.Windows.Data;
+#endif
+
+namespace Prism
+{
+    internal static class DependencyObjectExtensions
+    {
+        public static bool CheckAccess(this DependencyObject instance)
+#if __WASM__
+            // This needs to evolve, once threading is supported
+            // See https://github.com/unoplatform/uno/issues/2302
+            => System.Threading.Thread.CurrentThread.ManagedThreadId == 1;
+#elif HAS_WINUI
+            => instance.Dispatcher.HasThreadAccess;
+#else
+            => instance.CheckAccess();
```

Comment:
I'm not sure why this should be added in WPF... I get it for UNO/WASM but it seems like this should only exist for those platforms since CheckAccess already exists in WPF

Reply:
That's fair. It's mostly a force of habit to ensure that is shows what the method was originally replacing. I'll move the whole method in an Uno specific file.